app/scripts/modules/core/account/account.service.js

```diff
@@ -94,6 +94,42 @@ module.exports = angular.module('spinnaker.core.account.service', [
       return deferred.promise;
     });
 
+    let getUniqueRegionsForAllAccounts = _.memoize((provider) => {
+      return getRegionsKeyedByAccount(provider)
+        .then(function(regionsByAccount) {
+          let regions = _.chain(regionsByAccount)
+            .values()
+            .map(acct => acct.regions)
+            .flatten()
+            .map(reg => reg.name)
+            .uniq()
+            .value();
+
+          return regions;
+        });
+    });
+
+    let getUniqueGceZonesForAllAccounts = _.memoize((provider) => {
+      return getRegionsKeyedByAccount(provider)
+        .then(function(regionsByAccount) {
+          return _.chain(regionsByAccount)
+            .values()
+            .map(acct => acct.regions)
+            .flatten()
+            .reduce((acc, obj) => {
+              Object.keys(obj).forEach((key) => {
+                if(acc[key]) {
+                  acc[key] = _.uniq(acc[key].concat(obj[key]));
+                } else {
+                  acc[key] = obj[key];
+                }
+              });
+              return acc;
+            }, {})
+            .value();
+        });
+    });
```

Comment:
Another option is to just move this method into a different service in the GCE module, which would get around the configuration and delegation.